### DISCUSSION TITLE
Welcome to PowerShell Discussions!

### DISCUSSION THREAD

**Discussion by SteveL-MSFT** [PowerShell/PowerShell#14607] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

In general, the intent of Discussions is to keep Issues about the code in this repo and have issues be actionable by the team or community members.  So general PowerShell questions, or if you're not sure if something is a bug can be discussed here.  PowerShell team members may not regularly participate in discussions, but feel free to @mention team members if you need one of them to weigh in on a topic.

Discussions on GitHub is an experiment for the PowerShell team, so we'll continue to evaluate how to use this feature (or not use it) pending community feedback.

**Comment by doctordns** [comment]
Excellent.

I do not know much about this feature of GitHub but it looks promising.

And anything to move tech support questions out of Issues is good.

**Reply by ngawangpaljor** [reply]
hi

**Comment by apetitjean** [comment]
Hi guys!

I very much like good old forums! ;-)

First question: I noticed this morning when I launched PowerShell 7.1.0 that 7.1.1 was available. I made the update but I cannot find any release notes anywhere. Furthermore 7.1.1 is not released yet on PowerShell's the Github landing page.

Is that normal? Where can I find the release notes?

Thanks
Arnaud

**Reply by pepoluan** [reply]
> I very much like good old forums! ;-)

For me this is even better ... much easier to have a threaded discussion on one particular post... like this one! 😄 

> First question: I noticed this morning when I launched PowerShell 7.1.0 that 7.1.1 was available. I made the update but I cannot find any release notes anywhere. Furthermore 7.1.1 is not released yet on PowerShell's the Github landing page.
> 
> Is that normal? Where can I find the release notes?

Probably because PowerShell itself directly checks for some metadata files in the repo, so when the repo is updated -- which needs to happen before a "release" can be, well, released -- PowerShell picks up on the new version already, while the maintainers were busy packaging 7.1.1

It's a matter of timing 😉

**Reply by Rhenry50** [reply]
C:\WINDOWS\system32> "C:\Windows\System32" Set-MpPreference -AttackSurface ReductionRules_Ids D3E037E1-3EB8-44C8-A917-57927947596D AttackSurfaceReductionRules_Actions Enabled
At line:1 char:23
+ "C:\Windows\System32" Set-MpPreference -AttackSurface ReductionRules_ ...
+                       ~~~~~~~~~~~~~~~~
Unexpected token 'Set-MpPreference' in expression or statement.
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : UnexpectedToken

**Reply by Rhenry50** [reply]
KEEP GETTING THIS ERROR MESSSAGE

**Reply by aa99g** [reply]
mon systeme window11 réclame une mise à jour de aka.ms/PSWindows. Je ne sais pas le faire

**Comment by mikelon** [comment]
Is there a command line to update to the latest version of powershell or is the only way to download and install an updated MSI/ZIP? I've tried searching but all I can find are commands to upgrade to 7 from older pre-installed versions of powershell.

**Reply by SteveL-MSFT** [reply]
See https://github.com/PowerShell/PowerShell/issues/14217

**Comment by doctordns** [comment]
Yes.
There is a script you can get from the GitHub site to do this. In my next book, I illustrate this - you can get my scripts from here: https://github.com/doctordns/PACKT-PS7/tree/master/scripts/Ch%201%20-%20Install-Config%20Pwsh.

This includes a script to install PowerSHell7 (run from Windows PowerShell!"), create profiles, install VS Code and more.;

There are one-liners floating around but my script is more step by step!

**Comment by mikelon** [comment]
Thanks, I was hoping there might be a built-in command but in the meantime
those options will do.

**Comment by DellRay71** [comment]
How can I find out which version I have?

**Reply by doctordns** [reply]
Type
```powershell
$PSVersionTable
```

**Reply by DellRay71** [reply]
Thanks

**Comment by DellRay71** [comment]
Cool, thanks for the message. I have version 5.1 installed. Should I download and install 7.1? Or am I ok with 5? Raymond Dunne     From: Thomas LeeSent: Thursday, August 12, 2021 01:21 AMTo: PowerShell/PowerShellCc: DellRay71; CommentSubject: Re: [PowerShell/PowerShell] Welcome to PowerShell Discussions! (#14607) Type$PSVersionTable—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by doctordns** [reply]
not sure how you created that response but I can don't read it! :-)

**Comment by EGOISTqqq** [comment]
Very good.But me don't English.hh

**Comment by DellRay71** [comment]
?

**Comment by mubasherusman** [comment]
I do not want to see this starting info every time I open the this shell. How to disable/customize this 
![image](https://user-images.githubusercontent.com/4014673/143180923-3b02d44d-6893-4b15-b108-e2a2b9ea447c.png)

**Reply by vexx32** [reply]
Run with `-nologo` :)

**Reply by iSazonov** [reply]
`cls` in profile :-)

**Reply by doctordns** [reply]
Personally, on my system, I DO want to see the logo. I have PowerShell 7.2, 7.3 previews, the daily build, and Windows PowerShell. All there for good reasons and it is really helpful to ensure I am opening the console I wanted! 

CLS in $Profile works a treat!

**Reply by rkeithhill** [reply]
Give 7.3.0-preview.4 a try:

<img width="206" alt="image" src="https://user-images.githubusercontent.com/5177512/170531219-60d50466-411c-4654-a7f3-374d75ea2c1e.png">

If you'd still like to see the PS version info but none of the other text.

**Comment by DevRehannaLP** [comment]
I will be happy to be a part of this project.  I may not have been doing programming for a long time however my involvement with PowerShell I've come to like them!
Reginald Leon Portee  April 14, 2024 2:12 AM

**Comment by Rashed4377** [comment]
Hi, Can I access AWS EC2 through PowerShell? (like gitbash)

**Reply by StevenBucher98** [reply]
I am not sure how the AWS EC2 interface works but PowerShell is a shell and if you have the right AWS commands installed you certainly can accesses your EC2 instance!

**Comment by dgiesselbach** [comment]
Do you know when the PS 5.1 will be completely replaced by the PS 7 in Windows 11? In other words, when will the PS 7 be integrated into the Windows base build?

PS 7 has so many advantages and features that simply don't exist in PS 5.

In terms of automation, it would help us enormously if all newer Windows systems had the PS7 integrated by default.

Are there any plans yet? :)

**Reply by rhubarb-geek-nz** [reply]
> PS 7 has so many advantages and features that simply don't exist in PS 5.

The downside of PS7 being delivered as part of Windows is it effectively ceases to be an open source project on Windows and you get the version that Microsoft deems appropriate. For instance I use PowerShell 7.2 because I don't want certain features from 7.3 onwards. If it was part of Windows I would lose the control to determine what software I run on my machine.

**Reply by gregor190** [reply]
yeah i think that bad i use linux i am an linux user a linuxser

**Comment by gregor190** [comment]
do you want me to make something with .ps1 .ps1m and .psd1 files

**Comment by adelcollab-sys** [comment]
I'am MANAGER OF ZEUS WHO WANT TO JOIN US ? I chose 25 people

**Comment by ktdrivers7** [comment]
Dear Microsoft Learn and PowerShell Teams,

I am writing to propose a comprehensive idea that addresses a recurring gap in how users learn, understand, and safely adopt PowerShell and Windows command-line tools.

This proposal is not about adding more documentation—Microsoft already excels at that—but about **transforming learning into an interactive, guided experience that lives inside the tools themselves**.

---

## 1. The underlying problem

PowerShell is one of Microsoft’s most powerful technologies, yet it is frequently misunderstood, underused, or misused.

From my experience—and from countless community discussions—the typical learning path looks like this:

* Users copy commands from forums or documentation
* Commands are executed without understanding objects, pipelines, or side effects
* Mistakes occur that can damage system configurations
* Users become frustrated or afraid of PowerShell

The core issue is **not lack of documentation**, but lack of *guided comprehension*.

Microsoft Learn and Docs explain *what* commands do, but not always *how* users should think while using them.

---

## 2. The current learning gap

At present, learning PowerShell involves:

* Reading documentation
* Manually experimenting
* Debugging errors after they happen

This creates a steep learning curve, especially for:

* Beginners
* Career switchers
* Non-developers
* Power users transitioning from GUI tools

As a result:

* Users memorize commands instead of understanding concepts
* Pipelines feel “magical” rather than logical
* PowerShell is perceived as dangerous instead of empowering

---

## 3. Proposed solution: an interactive PowerShell learning mode

I propose an **interactive, guided learning environment for PowerShell**, conceptually similar to mobile learning apps like MIMO, but technically aligned with Microsoft Learn and Windows.

### Core characteristics:

* Runs **inside PowerShell or Windows Terminal**
* Structured as progressive lessons (Lesson 1, Lesson 2, etc.)
* Requires users to **type commands themselves**
* Validates results and outputs in real time
* Explains *why* a command works or fails
* Uses real PowerShell objects, not simulations

### Example interactions:

* Explaining what `Get-Location` returns and its object type
* Demonstrating how `Get-Member` reveals object capabilities
* Showing how pipelines transform objects step by step
* Explaining parameters through cause-and-effect, not syntax lists

---

## 4. Educational benefits for users

Such a system would:

* Build correct mental models from day one
* Reduce fear of the command line
* Encourage exploration instead of copy-paste
* Prevent common beginner mistakes
* Make PowerShell approachable for a wider audience

It would also naturally guide users through:

* Command Prompt basics
* PowerShell fundamentals
* Advanced scripting concepts
* Eventually, automation and DevOps workflows

---

## 5. Technical and operational benefits for Microsoft

From Microsoft’s perspective, this approach could:

* Reduce support incidents caused by incorrect command usage
* Decrease configuration errors on Windows systems
* Improve overall PowerShell adoption and reputation
* Strengthen Microsoft Learn as a best-in-class learning platform
* Create a safer, more educated user base

In short: **better-trained users create fewer problems and greater trust**.

---

## 6. Why PowerShell is uniquely suited for this

PowerShell already includes:

* A rich object-oriented pipeline
* Built-in discovery tools (`Get-Help`, `Get-Member`)
* Deep Windows integration
* A strong open-source ecosystem

What it lacks is a **didactic layer**—a way for the shell to actively teach while being used.

This proposal does not replace documentation; it **activates it**.

---

## 7. Potential challenges and drawbacks (acknowledged)

No proposal is without trade-offs. Potential concerns include:

* Development and maintenance cost
* Risk of oversimplifying advanced concepts
* Need for careful sandboxing to avoid system damage
* Ensuring lessons stay aligned with evolving PowerShell versions

However, these challenges are manageable:

* Lessons can run in constrained environments
* Advanced features can be opt-in
* Content can be modular and versioned
* Community feedback can guide iteration

---

## 8. Why this is worth doing

PowerShell is not just a tool—it is a gateway to understanding Windows itself.

By turning PowerShell into a **teacher**, Microsoft would:

* Lower the barrier to entry
* Increase long-term adoption
* Reduce misuse
* Empower users instead of intimidating them

This aligns perfectly with Microsoft’s broader goals of education, accessibility, and developer empowerment.

---

## Closing

I believe an interactive PowerShell learning experience—built on top of Microsoft Learn and Docs and integrated directly into Windows—would be a meaningful and forward-looking addition to the ecosystem.

Thank you for taking the time to read this proposal. I hope it contributes to ongoing discussions about how users learn and grow with Microsoft technologies.

Kind regards,
Ken



Hola equipos de Microsoft Learn y PowerShell,

Os escribo para proponer una idea que aborda un problema recurrente en la forma en que los usuarios aprenden, entienden y utilizan PowerShell y las herramientas de línea de comandos de Windows.

No se trata de añadir más documentación—Microsoft ya destaca en eso—sino de **transformar el aprendizaje en una experiencia interactiva y guiada integrada en las propias herramientas**.

---

### El problema de fondo

PowerShell es una de las tecnologías más potentes de Microsoft, pero a menudo se aprende mal:

* Copiando comandos sin entenderlos
* Ejecutándolos sin conocer los efectos reales
* Provocando errores de configuración
* Generando miedo o rechazo a la consola

El problema **no es la falta de información**, sino la falta de **comprensión guiada**.

---

### La brecha actual de aprendizaje

Hoy en día aprender PowerShell implica:

* Leer documentación
* Probar a ciegas
* Corregir errores después del daño

Esto crea una barrera grande para principiantes, usuarios no técnicos o personas que vienen del entorno gráfico.

---

### La solución propuesta

Un **modo de aprendizaje interactivo de PowerShell**, integrado con Microsoft Learn, que:

* Se ejecute dentro de PowerShell o Windows Terminal
* Esté estructurado por lecciones
* Obligue al usuario a escribir comandos
* Valide resultados en tiempo real
* Explique por qué algo funciona o falla
* Enseñe objetos, pipelines y lógica, no solo sintaxis

---

### Beneficios para los usuarios

* Aprendizaje real, no memorización
* Menos miedo a romper cosas
* Menos errores graves
* Mejor comprensión del sistema
* Evolución natural hacia scripting y automatización

---

### Beneficios para Microsoft

* Menos incidencias de soporte
* Menos errores por uso incorrecto
* Mejor adopción de PowerShell
* Usuarios más capacitados
* Refuerzo de Microsoft Learn como plataforma educativa

---

### Riesgos y desventajas (reales)

* Coste de desarrollo
* Necesidad de sandboxing
* Riesgo de simplificación excesiva
* Mantenimiento de contenidos

Pero todos estos riesgos son **gestionables** y asumibles frente a los beneficios.

---

### Por qué merece la pena

PowerShell no es solo una consola: es una forma de entender Windows.

Convertirlo en un entorno que **enseña mientras se usa** reduciría errores, frustración y barreras de entrada, y empoderaría a los usuarios.

---

Un saludo y gracias por vuestro tiempo,
Ken

**Reply by iSazonov** [reply]
@ktdrivers7 Best place for your proposal is https://github.com/MicrosoftDocs/PowerShell-Docs

**Reply by ktdrivers7** [reply]
Thanks! That makes sense. I’ll open an issue in MicrosoftDocs/PowerShell-Docs with the full proposal (CMD → PowerShell → Python interactive learning mode) and link it back here so it’s tracked in the right place.
